src/Sarif/Core/Tool.cs

```diff
@@ -49,5 +50,16 @@ public static Tool CreateFromAssemblyData(string prereleaseInfo = null)
 
             return tool;
         }
+
+        public bool ShouldSerializeRulesMetadata()
+        {
+            return this.RulesMetadata != null &&
+                    this.RulesMetadata.Where((n) => { return n != null; }).Any();
```

Comment:
>Where((n) => { return n != null; }).Any() [](start = 39, length = 41)

The second line can simplify to `this.RulesMetadata.Any(n => n != null)`. Taking it a step further, the whole method can collapse to
```
    return this.RulesMetadata?.Any(n => n != null) == true;
```

Reply:
No sir, the Any extension is associated with bool not Nullable<bool>. Here's the most concise form I've come up with, building on your idea.

return this.RulesMetadata != null && this.RulesMetadata.Any(n => n != null);

---
In reply to: [255306805](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255306805) [](ancestors = 255306805)